### ISSUE TITLE
test_guided_alignment_transformer.sh failed-std::exception in sub-thread: stoi.

### ISSUE THREAD

**Issue by wangxw1023** [marian-nmt/marian-dev#651]
### Bug description
I want to train a transformer model using the guided-alignment model.

At first, I use my own corpus and the parameter in test_guided_alignment_transformer.sh.
I meet the error: Error: Caught std::exception in sub-thread: stoi.

### How to reproduce
Then I tried to run your test_guided_alignment_transformer.sh with your corpus and model.
I also meet the same error. Caught std::exception in sub-thread: stoi. 
the entire log is in the attachment.

### Context
* Marian version: Paste the output of `--version` here
* CMake command: Type the cmake command you used and attach the output of `--build-info all`
* Log file: Attach your training/decoding logs 
[marian-guide-alignment-error-log.txt](https://github.com/marian-nmt/marian-dev/files/4603123/marian-guide-alignment-error-log.txt)

my marian-dev version is download from github in April 2020. I know you have said your regression test is passed in current version, so I'm really confused why I meet this problem.

Add any other information about the problem here.

If you need any other information, please contact me. Thanks very much. Look forward to your reply.

**Comment by snukky** [comment]
Are you able to use the same training command without guided-alignment with no similar issue?
How do you compile Marian and which GCC version do you use?

**Comment by wangxw1023** [comment]
> Are you able to use the same training command without guided-alignment with no similar issue?
> How do you compile Marian and which GCC version do you use?

I can use the same training command without guided-alignment and my gcc version is 5.4.0

**Comment by wangxw1023** [comment]
@snukky Hi, did you have any idea for my error? I really appreciate it if you can give me some advice.

**Comment by snukky** [comment]
Your word alignment is incorrectly generated for the given vocab. Make sure there is no indices exceeding the sentence length in the alignment file. Unfortunately, Marian does not check if word alignments are correct.

As this is an example from the regression tests, but run outside the test framework, I think what has happened here is you used wrong vocabs, hence a mismatch with the provided guided alignment file.
From your logs:
```
[2020-05-09 09:12:47] [data] Loading vocabulary from JSON/Yaml file vocab.en.yml
[2020-05-09 09:12:47] [data] Setting vocabulary size for input 0 to 16,082
[2020-05-09 09:12:47] [data] Loading vocabulary from JSON/Yaml file vocab.de.yml
[2020-05-09 09:12:48] [data] Setting vocabulary size for input 1 to 24,282
```
From the test output:
```
[2020-05-17 12:11:02] [data] Loading vocabulary from JSON/Yaml file vocab.en.yml
[2020-05-17 12:11:02] [data] Setting vocabulary size for input 0 to 16,538
[2020-05-17 12:11:02] [data] Loading vocabulary from JSON/Yaml file vocab.de.yml
[2020-05-17 12:11:02] [data] Setting vocabulary size for input 1 to 25,808
```

It should work if you run the regression test using (and following the instructions from [README](https://github.com/marian-nmt/marian-regression-tests) first):
```
bash run_mrt.sh tests/training/features/guided-alignment/test_guided_alignment_transformer.sh
```

Word alignments need to be generated from subword-segmented corpus, so they depend on vocabs.

**Comment by wangxw1023** [comment]
@snukky Thanks and I will try again.

**Comment by wangxw1023** [comment]
@snukky Hi, I generated the corpus.align file again and the test_guided_alignment_transformer.sh can test OK. I think the reason is I used one of my local models to generate the corpus.align, though I use the right vocab. When it comes to the local guided alignment, may due to the vocab is in one yml file, now I have separate the vocab into two files and try again.
Thank you very much for your help.
